app/controllers/kaui/subscriptions_controller.rb

```diff
@@ -118,8 +118,12 @@ def destroy
       # START_OF_TERM is *not* a valid entitlement_policy and so would default to IMMEDIATE
       entitlement_policy = billing_policy && billing_policy == 'START_OF_TERM' ? 'IMMEDIATE' : billing_policy
 
-      # true by default
-      use_requested_date_for_billing = (params[:use_requested_date_for_billing] || '1') == '1'
+      # true by default except default policy
+      use_requested_date_for_billing = if requested_date || billing_policy
+                                         (params[:use_requested_date_for_billing] || '1') == '1'
+                                       else
+                                         params[:use_requested_date_for_billing] == '1'
```

Comment:
`use_requested_date_for_billing` should be left `nil` here if `params[:use_requested_date_for_billing]` isn't present (to use the default behavior server side).

Reply:
If the user hasn't explicitly set `use_requested_date_for_billing`, then let's pass `nil` to the backend, so that the _defaults_ kick-in.

Reply:
We already have a test for this destroy method here: https://github.com/killbill/killbill-admin-ui/blob/79c9db06534cbfe3b2e51d71cbce0cfab0d74fc9/test/functional/kaui/subscriptions_controller_test.rb#L139
Do you mean to make another unit test for Kaui::Subscription?

Reply:
We can add another functional test, without passing `use_requested_date_for_billing` like the existing one.

Doing the exact scenario from https://github.com/killbill/killbill-admin-ui/issues/285 5 times would be ideal as a regression test. For each test, upload the catalog from the ticket, and verify each behavior from the drop down (you can retrieve the subscription after calling destroy to check its state).